### ISSUE TITLE
[deps] Choose between antd/icons and react-icons

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger-ui#1723]
Follow up on this discussion https://github.com/jaegertracing/jaeger-ui/pull/1721#issuecomment-1688653824

We are probably using about 10-20 different icons in the project, pulled rather randomly from antd and react-icons (and within react-icons from completely different sets).

The proposal is to consolidate onto a single set, for example react-icons/fa (assuming it can accommodate our existing usage).

* [ ] run grep/awk to come up with an inventory of all icons in use
* [ ] compare and select a single set that can represent them with equivalent icons
* [ ] make code changes
* [ ] remove unused dependencies

**Comment by yurishkuro** [comment]
I tagged this as `good first issue` because individual steps could be considered as such.

**Comment by Mohith234** [comment]
I would like to give it a try @yurishkuro. I'm new to this project and found this good first issue.

**Comment by Kavinjsir** [comment]
Trying to follow the instructions to list the icons imported from both libraries:
(Numbers are the frequency of each icon)

| `@ant-design/icons`      | `react-icons`          |
| ------------------------ | ---------------------- |
| 2 ClockCircleOutlined    | 2 FaCheck              |
| 5 CloseOutlined          | 3 FaFilter             |
| 1 DoubleRightOutlined    | 2 FaTrash              |
| 3 DownOutlined           | 4 IoAlert              |
| 2 ExportOutlined         | 2 IoAndroidArrowBack   |
| 1 FileAddOutlined        | 4 IoAndroidLocate      |
| 2 FilterOutlined         | 2 IoAndroidOpen        |
| 1 InfoCircleOutlined     | 2 IoArrowRightA        |
| 3 LoadingOutlined        | 7 IoChevronDown        |
| 1 ProfileOutlined        | 5 IoChevronRight       |
| 1 QuestionCircleOutlined | 3 IoHelp               |
| 1 RightOutlined          | 11 IoIosArrowDown      |
| 2 SearchOutlined         | 8 IoIosArrowRight      |
| 1 SmileFilled            | 2 IoIosFilingOutline   |
| 1 SmileOutlined          | 2 IoIosSearch          |
| 1 SmileTwoTone           | 2 IoNetwork            |
|                          | 2 MdExitToApp          |
|                          | 2 MdFileUpload         |
|                          | 2 MdInfoOutline        |
|                          | 2 MdKeyboardArrowDown  |
|                          | 2 MdKeyboardArrowRight |
|                          | 2 MdKeyboardArrowRight |
|                          | 3 MdVisibility         |
|                          | 5 MdVisibilityOff      |
|                          | 2 SortAmountAsc        |
|                          | 2 TiCancel             |

---

It feels like `react-icons` are comparatively used more frequently. Well, I'm curious if we can find a smooth way to substitute icons from either side..

---

FYI, below are the commands I'm trying to append with `grep` and `awk`:
(Plz fix me if I'm writing wrong 🙏)

For `@ant-design/icons`:

```shell
grep -r "from '@ant-design/icons'" ./packages/jaeger-ui | awk -F '{|}' '{ gsub(/ /, "", $2); split($2, a, ","); for(i in a) print a[i] }' | sort | uniq -c
```

For `react-icons`:
```shell
grep -r "from 'react-icons" ./packages/jaeger-ui | awk -F ' ' '{print $2}' | sort | uniq -c
```

**Comment by yurishkuro** [comment]
Pretty sure we're not using Smile. You need to grep in `src`, e.g. `./packages/jaeger-ui/src`, otherwise you're also including dependencies in node_modules (it's unlikely we render things with icons from those, and even if we do they are not in scope of this ticket).

You also need to include `plexus`, not just `jaeger-ui`

**Comment by Kavinjsir** [comment]
> 

Thx for pointing it out! Just adjusted the commands to focus on code in `jaeger-ui` and `plexus`:
```shell
# @ant-design/icons
grep -r "from '@ant-design/icons'" ./packages/jaeger-ui/src ./packages/plexus/src | awk -F '{|}' '{ gsub(/ /, "", $2); split($2, a, ","); for(i in a) print a[i] }' | sort | uniq -c
```

```shell
# react-icons
grep -r "from 'react-icons" ./packages/jaeger-ui/src ./packages/plexus/src | awk -F ' ' '{print $2}' | sort | uniq -c
```

Here is the result I got:

| `@ant-design/icons`      | `react-icons`          |
| ------------------------ | ---------------------- |
| 2 ClockCircleOutlined    | 1 FaCheck              |
| 5 CloseOutlined          | 2 FaFilter             |
| 1 DoubleRightOutlined    | 1 FaTrash              |
| 3 DownOutlined           | 2 IoAlert              |
| 2 ExportOutlined         | 1 IoAndroidArrowBack   |
| 1 FileAddOutlined        | 2 IoAndroidLocate      |
| 2 FilterOutlined         | 1 IoAndroidOpen        |
| 1 InfoCircleOutlined     | 1 IoArrowRightA        |
| 3 LoadingOutlined        | 4 IoChevronDown        |
| 1 ProfileOutlined        | 3 IoChevronRight       |
| 1 QuestionCircleOutlined | 2 IoHelp               |
| 1 RightOutlined          | 6 IoIosArrowDown       |
| 2 SearchOutlined         | 4 IoIosArrowRight      |
|                          | 1 IoIosFilingOutline   |
|                          | 1 IoIosSearch          |
|                          | 1 IoNetwork            |
|                          | 1 MdExitToApp          |
|                          | 1 MdFileUpload         |
|                          | 1 MdInfoOutline        |
|                          | 1 MdKeyboardArrowDown  |
|                          | 1 MdKeyboardArrowRight |
|                          | 2 MdVisibility         |
|                          | 3 MdVisibilityOff      |
|                          | 1 SortAmountAsc        |
|                          | 1 TiCancel             |

**Comment by yurishkuro** [comment]
@Kavinjsir for `react-icons`, I suggest including the namespace, e.g. `io/IoAndroidLocate` vs. `io5/IoAndroidLocate`, since they have the same names

The `io`/`io5` sets seem the most promising. If we can map all other icons into one of them, we'd have a concrete plan how to proceed.

**Comment by yurishkuro** [comment]
My suggesting would be to converge all icons on react-icons/io family, since this is the family already used the most in Jaeger.

**Comment by priyanshu-kun** [comment]
@yurishkuro  I create a document which contains all icons set that are being used in code base and their io5 alternatives as well. But there are some icons  that I'm not able to find in io5 set, Could you please take a look upon the doc?

https://docs.google.com/document/d/15KcGEMfOCl1AmyhMI5T0EuYYwLZz-pGEibiU2uS87Dw/edit?usp=sharing

**Comment by yurishkuro** [comment]
I don't quite follow the layout of your doc, esp. when some rows have more than 2 icons. It would be better to have these columns:
  1. icon name
  2. library / family
  3. picture
  4. corresponding io5 picture or NOT FOUND

**Comment by yurishkuro** [comment]
the icon name is especially important as otherwise it's very difficult to make your doc actionable - can't search the code for a picture.

**Comment by priyanshu-kun** [comment]
Got it! Let me fix that.

**Comment by priyanshu-kun** [comment]
@yurishkuro Now I provide as much details as I can, But the table stretch quite a bit but the table make sense now. I put icons in the columns to their corresponding name.
https://docs.google.com/document/d/15KcGEMfOCl1AmyhMI5T0EuYYwLZz-pGEibiU2uS87Dw/edit?usp=sharing

I hope it make sense to you as well

**Comment by yurishkuro** [comment]
@priyanshu-kun looks good (I don't think you needed so many columns in the table, the current family is obvious from the icon name). I left questions on some icons.

Do you want to start making changes by swapping the icons where there's no concern?

**Comment by priyanshu-kun** [comment]
@yurishkuro Yeah I would love do that!

**Comment by yurishkuro** [comment]
@priyanshu-kun after #1771 merged, I am seeing these as remaining places. Does that match your understanding?
```
$ grx "'@ant-design/icons'" packages/*/src/
packages/jaeger-ui/src//components/common/LoadingIndicator.tsx:16:import { LoadingOutlined } from '@ant-design/icons';
packages/jaeger-ui/src//components/common/CopyIcon.tsx:23:import { CopyOutlined } from '@ant-design/icons';
packages/jaeger-ui/src//components/TracePage/ArchiveNotifier/index.tsx:17:import { LoadingOutlined } from '@ant-design/icons';
packages/jaeger-ui/src//components/TracePage/ArchiveNotifier/index.test.js:18:import { LoadingOutlined } from '@ant-design/icons';
packages/jaeger-ui/src//components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx:18:import { LinkOutlined } from '@ant-design/icons';
packages/jaeger-ui/src//components/TracePage/TraceTimelineViewer/TimelineHeaderRow/TimelineCollapser.tsx:17:import { DoubleRightOutlined } from '@ant-design/icons';
```

**Comment by yurishkuro** [comment]
Let's pick another family then for the remaining ones, preferably with MIT license.
